src/map.cpp

```diff
@@ -10124,85 +10131,82 @@ const pathfinding_cache &map::get_pathfinding_cache_ref( int zlev ) const
     return cache;
 }
 
-void map::update_pathfinding_cache( int zlev ) const
+void map::update_pathfinding_cache( const tripoint &p ) const
```

Comment:
This looks like a lot of code changed, but all this is is lifting the inner loop out of `update_pathfinding_cache(int zlev)` into a per tile method.